### PR TITLE
Make new zsh completion #132

### PR DESCRIPTION
**Pull request by Turiok**
Hello everyone,

I have succeeded with the old zsh completion.
Adding in my .zshrc 
```
autoload -U +X bashcompinit && bashcompinit
source /path/to/your/gita_completion_script
``` 
But the completion is not Zsh style and seems very limited.

I propose a new zsh completion.
It's my second completion zsh script. So be nice and give me pointers if I'm wrong.

I find the completion nicer and dynamic depending your configuration. By exemple, it display your repositories or your groups.

To me, it miss 2 things : 
- Completion for color command `gita color set <TAB>` displaying the possible colors.
- For the `gita group add` command. I don't know if options `-n` and `-p` are exclusive? Easy to correct. I just have to know

Other problem. If the community doesn't know zsh completion. It is probably complex to maintain.
Maybe, I'll propose this completion to other repository like https://github.com/zsh-users/zsh-completions if @nosarthur doesn't want this PR.

### CODE REVIEW

**Comment by nosarthur** [comment]
Thanks for the contribution. Maybe add a folder and place both completion script there so people can choose?

**Comment by Turiok** [comment]
Do you want I move the bash completion version in bash folder?

**Comment by nosarthur** [comment]
sure, thanks

One minor suggestion: the folder structure can be simplified to `auto-completion/zsh/...` and `auto-completion/bash/...`

**Comment by Turiok** [comment]
OK to change the folder. I made  that in case you have other contribution for other people.

File: README.md [hunk 1]
```diff
@@ -158,12 +158,16 @@ the branch color to work.
 See [this stackoverflow post](https://stackoverflow.com/questions/51680709/colored-text-output-in-powershell-console-using-ansi-vt100-codes) for details.
 
 ## Auto-completion
+### Bash
+Download [.gita-completion.bash](https://github.com/nosarthur/gita/blob/master/.gita-completion.bash) and source it in shell.
+
+### Zsh
+There is 2 options :
```
Comment (nosarthur) [inline README.md:165]: is => are

File: gita/__main__.py [hunk 2]
```diff
@@ -780,17 +780,18 @@ def main(argv=None):
     cmds = utils.get_cmds_from_files()
     for name, data in cmds.items():
         help = data.get("help")
+        repo_help = help
```
Comment (nosarthur) [inline gita/__main__.py:783]: why split `help` into two?

**Review by nosarthur — CHANGES_REQUESTED** [review]

File: gita/__main__.py (hunk 2 above)
Comment (Turiok) [reply gita/__main__.py:783]: With the same help than before, I think the general help is too loaded.
But I can go back

<details>
  <summary>Before</summary>
The general help : 
gita -h

``` bash
sub-commands:
  {add,rm,freeze,clone,rename,flags,color,info,ll,context,ls,group,super,shell,clear,br,clean,diff,difftool,fetch,last,lo,log,merge,mergetool,patch,pull,push,rebase,reflog,remote,reset,show,stash,stat,st,tag}
                        additional help with sub-command -h
    add                 add repo(s)
    rm                  remove repo(s)
    freeze              print all repo information
    clone               clone repos
    rename              rename a repo
    flags               git flags configuration
    color               color configuration
    info                information setting
    ll                  display summary of all repos
    context             set context
    ls                  show repo(s) or repo path
    group               group repos
    super               run any git command/alias
    shell               run any shell command
    clear               removes all groups and repositories
    br                  show local branches for the chosen repo(s) or group(s)
    clean               remove all untracked files/folders for the chosen repo(s) or group(s)
    diff                git show differences for the chosen repo(s) or group(s)
    difftool            show differences using a tool for the chosen repo(s) or group(s)
    fetch               fetch remote update for all repos or for the chosen repo(s) or group(s)
    last                show log information of HEAD for the chosen repo(s) or group(s)
    lo                  show one-line log for the latest 7 commits for all repos or for the chosen repo(s) or group(s)
    log                 show logs for the chosen repo(s) or group(s)
    merge               merge remote updates for the chosen repo(s) or group(s)
    mergetool           merge updates with a tool for the chosen repo(s) or group(s)
    patch               make a patch for the chosen repo(s) or group(s)
    pull                pull remote updates for all repos or for the chosen repo(s) or group(s)
    push                push the local updates for all repos or for the chosen repo(s) or group(s)
    rebase              rebase from master for the chosen repo(s) or group(s)
    reflog              show ref logs for the chosen repo(s) or group(s)
    remote              show remote settings for the chosen repo(s) or group(s)
    reset               reset repo(s) for the chosen repo(s) or group(s)
    show                show detailed commit information for the chosen repo(s) or group(s)
    stash               store uncommited changes for the chosen repo(s) or group(s)
    stat                show edit statistics for all repos or for the chosen repo(s) or group(s)
    st                  show status for all repos or for the chosen repo(s) or group(s)
    tag                 show tags for the chosen repo(s) or group(s)
``` 

a specific help : 
gita st -h : 

``` bash
usage: gita st [-h] [-s SHELL] [{gita,artifactory-cleanup,arch-mu-repo,stackedit-docker,gitlab-branch-source-plugin,analyse_bonnes_pratiques,ul_cnes,open_source,} ...]

show status for all repos or for the chosen repo(s) or group(s)

positional arguments:
  {gita,artifactory-cleanup,arch-mu-repo,stackedit-docker,gitlab-branch-source-plugin,analyse_bonnes_pratiques,ul_cnes,open_source,[]}
                        show status for all repos or for the chosen repo(s) or group(s)

options:
  -h, --help            show this help message and exit
  -s SHELL, --shell SHELL
                        If set, run in shell mode
``` 
</details>

with my version : 

<details>
  <summary>After</summary>
general help : 
gita -h

``` bash
sub-commands:
  {add,rm,freeze,clone,rename,flags,color,info,ll,context,ls,group,super,shell,clear,br,clean,diff,difftool,fetch,last,lo,log,merge,mergetool,patch,pull,push,rebase,reflog,remote,reset,show,stash,stat,st,tag}
                        additional help with sub-command -h
    add                 add repo(s)
    rm                  remove repo(s)
    freeze              print all repo information
    clone               clone repos
    rename              rename a repo
    flags               git flags configuration
    color               color configuration
    info                information setting
    ll                  display summary of all repos
    context             set context
    ls                  show repo(s) or repo path
    group               group repos
    super               run any git command/alias
    shell               run any shell command
    clear               removes all groups and repositories
    br                  show local branches
    clean               remove all untracked files/folders
    diff                git show differences
    difftool            show differences using a tool
    fetch               fetch remote update
    last                show log information of HEAD
    lo                  show one-line log for the latest 7 commits
    log                 show logs
    merge               merge remote updates
    mergetool           merge updates with a tool
    patch               make a patch
    pull                pull remote updates
    push                push the local updates
    rebase              rebase from master
    reflog              show ref logs
    remote              show remote settings
    reset               reset repo(s)
    show                show detailed commit information
    stash               store uncommited changes
    stat                show edit statistics
    st                  show status
    tag                 show tags
``` 


a specific help : 
gita st -h : 

``` bash
usage: gita st [-h] [-s SHELL] [{gita,artifactory-cleanup,arch-mu-repo,stackedit-docker,gitlab-branch-source-plugin,analyse_bonnes_pratiques,ul_cnes,open_source,} ...]

show status

positional arguments:
  {gita,artifactory-cleanup,arch-mu-repo,stackedit-docker,gitlab-branch-source-plugin,analyse_bonnes_pratiques,ul_cnes,open_source,[]}
                        show status for all repos or for the chosen repo(s) or group(s)

options:
  -h, --help            show this help message and exit
  -s SHELL, --shell SHELL
                        If set, run in shell mode

``` 
</details>

Comment (nosarthur) [reply gita/__main__.py:783]: thanks, this is great

**Review by nosarthur — APPROVED** [review]

**Pull request merged**